Add DMs on infractions

Sem1083 opened the pull request:
DM_ON_ACTIONS

Sem1083 committed:
KICK

Sem1083 committed:
BAN

Sem1083 committed:
Mute

Sem1083 committed:
_UNMUTE

Sem1083 committed:
MUNMUTE

Sem1083 committed:
MUTE UPDATE

Sem1083 committed:
Indent

Sem1083 committed:
Warn

Sem1083 committed:
Configure Commands

Sem1083 committed:
DM_ACTION

Sem1083 committed:
Added certain logs

Sem1083 committed:
Added Migration

Sem1083 committed:
Added infractions

Sem1083 committed:
Added needed phrases

Sem1083 committed:
typo

Sem1083 committed:
Added Tempban

Sem1083 committed:
Added Tempban

Sem1083 committed:
Added Tempban | V26

Sem1083 committed:
Added Tempban

Sem1083 committed:
Added Tempban

AEnterprise commented on GearBot/Cogs/Moderation.py:
```diff
@@ -205,8 +205,18 @@ async def kick(self, ctx, user: discord.Member, *, reason: Reason = ""):
-
-    async def _kick(self, ctx, user, reason, message):
+        
+        name = Utils.clean_user(user)
```
why is this here? now regular kick will result in you getting the DM twice, because the line above will cause it to call the _kick function (and you'll even get a DM if they try to kick someone who you can't kick)

Sem1083 replied:
Fixed.

AEnterprise commented on GearBot/Cogs/Moderation.py:
```diff
@@ -281,9 +302,23 @@ async def bean(self, ctx, user: discord.Member, *, reason: Reason = ""):
             member = ctx.guild.get_member(user.id)
             await self._ban_command(ctx, member, reason, 0)
+
+        name = Utils.clean_user(user)
```
same as above

AEnterprise commented on GearBot/Cogs/Moderation.py:
```diff
@@ -330,11 +365,23 @@ async def tempban(self, ctx: commands.Context, user: DiscordUser, duration: Dura
             duration_seconds = duration.to_seconds(ctx)
             if duration_seconds > 0:
-
+                name = Utils.clean_user(user)
```
again here

Sem1083 replied:
fixed

AEnterprise commented on lang/en_US.json:
```diff
@@ -264,9 +264,32 @@
   "censor_trusted_bypass": "Censor Trusted Bypass has been {status}",
   "censor_trusted_bypass_unchanged": "Trusted Censor Bypass is already {status}",
   "warning_dm": "You have been warned in **{server}**!",
+  "kick_dm": "You have been kicked in **{server}**!",
```
might be worth changing the wording in these a little bit. like "You have been kicked in _server_ for ..."

Sem1083 replied:
done

AEnterprise reviewed (changes requested):
some issues below, also this doesn't actually include the reasons in an obvious way

Sem1083 committed:
Removed the DM from kick and ban functions.

Also moved the DMing to the top for tempban, _ban and _kick

Sem1083 committed:
Changed wording. 

Changed wording for warning_dm, kick_dm, ban_dm, unmute_dm, mute_dm, tempban_dm, extend_mute_dm, mute_duration_change_dm and mute_duration_until_dm to make it more understandable.

AEnterprise reviewed (changes requested):
tested and still has some issues:

- missing help string for the configuration commands
- unmute does not work (even with DM on unmute disabled, see screenshot)


![image](https://user-images.githubusercontent.com/5206666/91656184-07437e80-eab7-11ea-8a2a-920308f9d498.png)

Sem1083 commented:
Seems like I used user instead of target, will fix in a good hour

Sem1083 committed:
Fixed unmute, changed user to target. 

This should make unmute functional

Sem1083 committed:
Added Configuration Help 

Did this for dm_on_tempban_help, dm_on_ban_help, dm_on_kick_help, dm_on_mute_help and dm_on_unmute_help
